server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/RenameAction.java

```diff
@@ -59,13 +80,36 @@ public void define(WebService.NewController controller) {
 
   @Override
   public void handle(Request request, Response response) throws Exception {
-    qProfileWsSupport.checkQProfileAdminPermission();
-
     String newName = request.mandatoryParam(PARAM_PROFILE_NAME);
     String profileKey = request.mandatoryParam(PARAM_PROFILE_KEY);
+    doHandle(newName, profileKey);
+    response.noContent();
+  }
 
-    profileFactory.rename(profileKey, newName);
+  @VisibleForTesting
+  void doHandle(String newName, String profileKey) {
+    checkRequest(StringUtils.isNotBlank(newName), "Name must be set");
+    checkRequest(newName.length() <= MAXIMUM_NAME_LENGTH, String.format("Name is too long (>%d characters)", MAXIMUM_NAME_LENGTH));
+    userSession.checkLoggedIn();
```

Comment:
you could even do this check right in the `handle` method
no need to process parameters if the user is not logged in